addons/controllers/packageinstallstatus_controller.go

```diff
@@ -384,49 +381,19 @@ func (r *PackageInstallStatusReconciler) processPkgiIsManagedAndStatusChanged(o
 		return false
 	}
 
-	clusterObjKey := r.getClusterNamespacedName(pkgi)
+	clusterObjKey := getClusterNamespacedName(pkgi)
 	if clusterObjKey == nil {
 		return false
 	}
 
-	isManaged, err := r.isPackageManaged(*clusterObjKey, pkgi.Name)
-	if err != nil {
-		// just log the error cause it is called from the predicate function with only bool return value
-		log.Error(err, "failed to determine whether the package is managed or not")
-		return false
-	}
-
-	return isManaged
-}
-
-// isPackageManaged checks if the provided PackageInstall is among the list of managed(core/additional) packages
-func (r *PackageInstallStatusReconciler) isPackageManaged(clusterObjKey client.ObjectKey, pkgiName string) (bool, error) {
```

Comment:
Until we can find another way to address checking if a package is managed, I am removing this because it requires a client and putting a stateful function (one that has a ref to reconciler) is contributing towards memory leaks.